tailor_distro/manage/compare.py

```diff
@@ -51,19 +52,42 @@ def get_diff(self, repo):
         else:
             diff['name'] = {'unchanged': repo}
 
-        for field in ['type', 'url', 'version']:
-            try:
-                upstream = self.upstream_distro.repositories[repo].source_repository.get_data().get(field, None)
-            except (KeyError, AttributeError):
-                upstream = None
-            try:
-                internal = self.internal_distro.repositories[repo].source_repository.get_data().get(field, None)
-            except (KeyError, AttributeError):
-                internal = None
-
-            if internal != upstream:
-                if internal is not None:
-                    diff[field]['internal'] = internal
-                if upstream is not None:
-                    diff[field]['upstream'] = upstream
+        diff_url = build_diff(
+            upstream=get_url(self.upstream_distro.repositories.get(repo)),
+            internal=get_url(self.internal_distro.repositories.get(repo)))
+
+        if diff_url:
+            diff['url'] = diff_url
```

Comment:
Previously, you would set the url to be the internal url if it existed, otherwise the upstream. Now you're setting it to the combo-upstream/interal dictionary. Is that the intended effect?